examples/3d/perspective_zoom.rs

```diff
@@ -0,0 +1,125 @@
+        .add_systems(Startup, (setup, instructions))
+        .add_systems(Update, camera_controls)
+        .register_type::<CameraSettings>()
+        .run();
+}
+
+/// Set up a simple 3D scene
+fn setup(
+    mut camera_settings: ResMut<CameraSettings>,
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    camera_settings.orbit_speed = 0.02;
+    // Clamp FOV to this range. Note that we can't adjust FOV to more than PI,
+    // which represents a 180 degree field.
+    camera_settings.zoom_range = 0.5..3.0;
+    camera_settings.zoom_speed = 1.0;
+
+    commands.spawn(Camera3dBundle {
+        transform: Transform::from_xyz(5.0, 5.0, 5.0).looking_at(Vec3::ZERO, Vec3::Y),
+        ..default()
+    });
+
+    // Plane
+    commands.spawn(PbrBundle {
+        mesh: meshes.add(Plane3d::default().mesh().size(5.0, 5.0)),
+        material: materials.add(Color::srgb(0.3, 0.5, 0.3)),
+        ..default()
+    });
+
+    // Cube
+    commands.spawn(PbrBundle {
+        mesh: meshes.add(Cuboid::default()),
+        material: materials.add(Color::srgb(0.8, 0.7, 0.6)),
+        transform: Transform::from_xyz(1.5, 0.5, 1.5),
+        ..default()
+    });
+
+    // Light
+    commands.spawn(PointLightBundle {
+        transform: Transform::from_xyz(3.0, 8.0, 5.0),
+        ..default()
+    });
+}
+
+fn instructions(mut commands: Commands) {
+    commands
+        .spawn(NodeBundle {
+            style: Style {
+                align_items: AlignItems::Start,
+                flex_direction: FlexDirection::Column,
+                justify_content: JustifyContent::Start,
+                width: Val::Percent(100.),
+                ..default()
+            },
+            ..default()
+        })
+        .with_children(|parent| {
+            parent.spawn(TextBundle::from_section(
+                "Scroll mouse wheel to zoom in/out",
+                TextStyle::default(),
+            ));
+            parent.spawn(TextBundle::from_section(
+                "A or D to orbit left or right",
+                TextStyle::default(),
+            ));
+        });
+}
+
+fn camera_controls(
+    mut camera: Query<(&mut Projection, &mut Transform), With<Camera>>,
+    camera_settings: Res<CameraSettings>,
+    keyboard_input: Res<ButtonInput<KeyCode>>,
+    mouse_wheel_input: Res<AccumulatedMouseScroll>,
+) {
+    let mut delta_orbit = 0.0;
+    if keyboard_input.pressed(KeyCode::KeyA) {
+        // Orbit left
+        delta_orbit -= camera_settings.orbit_speed;
+    }
+    if keyboard_input.pressed(KeyCode::KeyD) {
+        // Orbit right
+        delta_orbit += camera_settings.orbit_speed;
+    }
+
+    let (mut projection, mut transform) = camera.single_mut();
+
+    if delta_orbit != 0.0 {
+        // Orbit the camera around a fixed point, facing its center.
+        transform.translate_around(Vec3::ZERO, Quat::from_axis_angle(Vec3::Y, delta_orbit));
```

Comment:
This is framerate-bound. I'm a bit worried that someone will copy-paste this into their code and have learned bad habits as a result.

Reply:
I know what some of those words mean.

Fortunately, I know where to find you on Discord if I get stuck :detective:

Reply:
Oh! Delta time. OK, I'm back.

Reply:
Haha, sorry for the obtuse language. Yeah, feel free to ping me if you need help :D